src/App.tsx

```diff
@@ -15,71 +15,89 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends Component {
+  state = {
+    selected: 'Jam',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  clearSelected = () => {
+    this.setState({ selected: '' });
+  }
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  selectItem = (itemName: string) => {
+    return () => this.setState({ selected: itemName })
+  }
```

Comment:
A bit more concise in my opinion.
```suggestion
  selectItem = (itemName: string) =>  () => {
  this.setState({ selected: itemName }) 
  }
```